source/module_esolver/esolver_ks_pw.cpp

```diff
@@ -306,15 +306,15 @@ namespace ModuleESolver
             // be careful that istep start from 0 and iter start from 1
             if((istep==0||istep==1)&&iter==1) 
             {
-                hsolver::DiagoIterAssist::need_subspace = false;
+                hsolver::DiagoIterAssist<double>::need_subspace = false;
```

Comment:
How about setting `double` as default type?

Reply:
Even if the default template parameter is set to double, one need to set the class call to `hsolver:: DiagoIterAssist<>:: need_ subspace = false` instead of `hsolver:: DiagoIterAssist:: need_ subspace = false`;  Still need to change the calling method of the class.